packages/shared/routes/dashboard/wallet/views/AccountHistory.svelte

```diff
@@ -72,6 +72,38 @@
             }
         }
     }
+
+    const filters = ['all', 'incoming', 'outgoing']
+    let activeFilterIndex = 0
+
+    let searchActive = false
+    $: searchValue = searchActive ? searchValue : ''
+
+    let filteredTransactions = transactions
+    $: switch (activeFilterIndex) {
+        case 0:
+            filteredTransactions = transactions
+            break
+        case 1:
+            filteredTransactions = transactions.filter(transaction => transaction.payload.data.essence.data.incoming)
```

Comment:
to avoid possible crashes might be good using `transaction?.payload?.data?.essence?.data?.incoming` everywhere